### ISSUE TITLE
ExternalLink component can't be used in save function

### ISSUE THREAD

**Issue by creativecoder** [WordPress/gutenberg#48807]
### Description

Using the ExternalLink component in the save function of a block results in an error and white screen when loading the editor.

The underlying issue appears to be https://github.com/WordPress/gutenberg/issues/43201.

### Step-by-step reproduction instructions

Create a new block, and use the `ExternalLink` component in the save function. When loading the editor, you'll get the following error.

```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
    Zf react-dom.js:1
    useContext react.js:1
    Yl context-system-provider.js:28
    Nu use-context-system.js:29
    Vu component.tsx:23
    q serialize.js:618
    Y serialize.js:734
    X serialize.js:659
```

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by creativecoder** [comment]
I'm not sure if this error is expected, or not, but I was disappointed I couldn't use the ExternalLink function in my block--it's quite useful to have the appropriate anchor attributes and accessibility text included.

**Comment by Mamaduka** [comment]
I don't believe this is a bug. Most components aren't meant to be used in the `save` function.

**Comment by talldan** [comment]
I agree with @Mamaduka, the components package is more intended for building UIs in the WordPress admin area rather than the frontend of a site, so I don't think it'd be considered a bug.

It's not to say that there couldn't be a low level package of pure components for that purpose, but that's not something being worked on in the gutenberg project.

**Comment by Mamaduka** [comment]
Closing as "won't fix". A "pure" external link can be implemented in userland.
